corehq/apps/domain/static/domain/js/encoder.js

```diff
@@ -0,0 +1,42 @@
+(function() {
+    function HexParsr() {
+        // private property
+        var _paddingLeft = "sha256$" + Math.random().toString(36).slice(2, 8);
+        var _paddingRight = Math.random().toString(36).slice(2, 8) + "=";
```

Comment:
It is possible that these `Math.random().toString(36)` calls will return a string that is shorter than 8 characters:
```
-> (0.0).toString(36)
<- "0"
```